numpyro/distributions/continuous.py

```diff
@@ -121,9 +121,10 @@ def sample(self, key, size=()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        normalize_term = (np.sum(gammaln(self.concentration), axis=-1) -
-                          gammaln(np.sum(self.concentration, axis=-1)))
-        return np.sum(np.log(value) * (self.concentration - 1.), axis=-1) - normalize_term
+        concentration = lax.convert_element_type(self.concentration, value.dtype)
+        normalize_term = (np.sum(gammaln(concentration), axis=-1) -
+                          gammaln(np.sum(concentration, axis=-1)))
```

Comment:
gammaln requires `concentration` is float, so we need to use `convert_element_type`. Value is a simplex so it should be `float`.

Reply:
I think we are stricter in terms of typing as compared to scipy distributions, and you'll probably find other distributions too which would throw an error if we passed in `np.array([1])` instead of `np.array([1.])` for some parameter (or value), when it expected a float. I think it would make sense to consistently handle this one way or the other. I prefer the stricter approach simply because its much easier to maintain than having to do this casting for all `sample` and `log_prob` methods. e.g. Another question might be whether we also want to support casting when a mix of float32 and float64 values are present, which would further complicate matters. 

If you think this will be nice to have, how about we place this logic in `__init__` and just convert it into the default float type (maybe we can disregard cases with mixed float32 and float64 types to begin with)? We can later handle this for other distributions.

Reply:
Until now, I only observe that gammaln requires float as an argument, so we need casting before applying this operator. I think that it is jax issue (not casting to canonical float for this operator) but it might be not necessary to raise an issue upstream. If there are other operators which needs this casting, we might consider making things consistence.